### PR TITLE
Add BashismsCheck from openSUSE/rpmlint-checks (#227).

### PR DESCRIPTION
**Pull request by marxin**

### CODE REVIEW

File: rpmlint/checks/BashismsCheck.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#############################################################################
+# File          : BashismsCheck.py
+# Purpose       : check /bin/sh shell scripts for bashisms
+#############################################################################
```
Comment (scarabeusiv) [inline rpmlint/checks/BashismsCheck.py]: The header is kinda how to say it, pointless.

File: rpmlint/checks/BashismsCheck.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    def __init__(self, config, output):
+        super().__init__(config, output, r'.*')
+
+    def check_file(self, pkg, filename):
+        root = pkg.dirName()
+        pkgfile = pkg.files()[filename]
+        fullpath = root + filename
+
+        if not (stat.S_ISREG(pkgfile.mode) and
+                pkgfile.magic.startswith('POSIX shell script')):
```
Comment (scarabeusiv) [inline rpmlint/checks/BashismsCheck.py]: Isn't 'POSIX shell script' everything that ends with .sh ?
We have to really implement queries for the pkg.py something like pkg.configs, pkg.headers, pkg.binaries,...

Comment (marxin) [reply rpmlint/checks/BashismsCheck.py]: > Isn't 'POSIX shell script' everything that ends with .sh ?

Nope:
```
$ file /usr/bin/* | grep 'POSIX shell script' | grep -v '\.sh' | wc -l
500
```

Comment (scarabeusiv) [reply rpmlint/checks/BashismsCheck.py]: Interesting, i am not disputing that but I created sample .sh file where I set shebang to point to various areas:
 * bash ` Bourne-Again shell script, ASCII text executable`
 * env sh `a /usr/bin/env sh script, ASCII text executable`
 * /bin/sh `POSIX shell script, ASCII text executable`
 * /usr/bin/sh `a /usr/bin/sh script, ASCII text executable`

So probably it is correct right?

Comment (marxin) [reply rpmlint/checks/BashismsCheck.py]: > So probably it is correct right?

I guess so.

File: test/test_bashisms.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+@pytest.mark.parametrize('package', ['binary/bashisms'])
+def test_bashisms(tmpdir, package, bashismscheck):
+    output, test = bashismscheck
+    test.check(get_tested_package(package, tmpdir))
+    out = output.print_results(output.results)
+    # TODO: enable once we fix checkbashisms on Leap 15
+    # which is broken due to a Perl regex error:
+    # https://build.opensuse.org/request/show/725994
+    # assert 'W: potential-bashisms /bin/script1' in out
```
Comment (scarabeusiv) [inline test/test_bashisms.py]: Just enable this, we will fix it SOON TM.

**Comment by marxin** [comment]
@scarabeusiv  Should I disable the check by default?

**Comment by scarabeusiv** [comment]
@marxin yes please this one should be off by default.

**Comment by marxin** [comment]
> @marxin yes please this one should be off by default.

Done.

**Pull request merged**